File: lista1/ocaml/zad3.ml
let rec insert (list, element) = 
  if list = [] then [(element, 1)]
  else
    let head = List.hd list in
    let tail = List.tl list in
    if fst head = element then (fst head, snd head + 1) :: tail
    else head :: insert (tail, element)
;;

insert [('a', 1); ('b', 2)] 'c';;
insert [('a', 1); ('b', 2)] 'b';;

File: lista1/scala/zad3.scala
def insert[A](list: List[(A, Int)], element: A): List[(A, Int)] =
  if (list == Nil) List((element, 1))
  else if (list.head._1 == element) (list.head._1, list.head._2 + 1) :: list.tail
  else list.head :: insert(list.tail, element)
  
insert(List(('a', 1), ('b', 2)), 'c')
insert(List(('a', 1), ('b', 2)), 'a')
